src/coreclr/nativeaot/System.Private.Reflection.Core/src/System/Reflection/Runtime/CustomAttributes/NativeFormat/NativeFormatCustomAttributeData.cs

```diff
@@ -189,11 +190,29 @@ internal sealed override IList<CustomAttributeNamedArgument> GetNamedArguments(b
                     Debug.Assert(!throwIfMissingMetadata);
                     return null;
                 }
-                customAttributeNamedArguments.Add(ReflectionAugments.CreateCustomAttributeNamedArgument(this.AttributeType, memberName, isField, typedValue));
+
+                customAttributeNamedArguments.Add(CreateCustomAttributeNamedArgument(this.AttributeType, memberName, isField, typedValue));
             }
             return customAttributeNamedArguments;
         }
 
+        [UnconditionalSuppressMessage("ReflectionAnalysis", "IL2080:UnrecognizedReflectionPattern",
+            Justification = "Metadata generation ensures fields/properties referenced from attributes are preserved.")]
+        private static CustomAttributeNamedArgument CreateCustomAttributeNamedArgument(Type attributeType, string memberName, bool isField, CustomAttributeTypedArgument typedValue)
```

Comment:
This will resolve the MemberInfo a bit earlier than before, but I do not think that it matters in practice. It is unlikely that somebody asks for CustomAttributeNamedArgument and won't look at the MemberInfo.

Reply:
They can look at `MemberName` instead. It's actually easier to match up that way if I were to crack CustomAttributeNamedArguments myself. But that's probably a perf optimization that should be done in the runtime repo first.